salt/utils/beacons.py

```diff
@@ -14,3 +14,13 @@ def remove_hidden_options(config, whitelist):
             if func.startswith("_") and func not in whitelist:
                 config.remove(entry)
     return config
+
+
+def list_to_dict(config):
+    """
+    Convert list based beacon configuration
+    into a dictionary.
+    """
+    _config = {}
+    list(map(_config.update, config))
```

Comment:
just a question here - do we know if this is faster than

```
for thing in config:
    _config.update(thing)
```
?

Or
```
[_config.update(x) for x in config]
```
I don't know if it's important, but seeing how many different places this is used, that information would at least be interesting!

Reply:
@waynew FYI.

In [23]: %timeit list(map(_config.update, config))
392 ns ± 1.79 ns per loop (mean ± std. dev. of 7 runs, 1000000 loops each)

In [28]: %timeit [_config.update(x) for x in config]
292 ns ± 2.38 ns per loop (mean ± std. dev. of 7 runs, 1000000 loops each)

In [30]: %%timeit
    ...: for thing in config:
    ...:     _config.update(thing)
    ...: 
151 ns ± 0.366 ns per loop (mean ± std. dev. of 7 runs, 10000000 loops each)